apps/opik-backend/src/main/java/com/comet/opik/domain/OpenTelemetryService.java

```diff
@@ -0,0 +1,84 @@
+package com.comet.opik.domain;
+
+import com.comet.opik.api.SpanBatch;
+import com.comet.opik.api.Trace;
+import com.comet.opik.utils.AsyncUtils;
+import com.google.inject.ImplementedBy;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+
+@ImplementedBy(OpenTelemetryServiceImpl.class)
+public interface OpenTelemetryService {
+
+    Mono<Long> parseAndStoreSpans(@NonNull ExportTraceServiceRequest traceRequest, @NonNull String projectName,
+            @NonNull String userName, @NonNull String workspaceName, @NonNull String workspaceId);
+}
+
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+@Slf4j
+class OpenTelemetryServiceImpl implements OpenTelemetryService {
+
+    private final @NonNull TraceService traceService;
+    private final @NonNull SpanService spanService;
+
+    @Override
+    public Mono<Long> parseAndStoreSpans(@NonNull ExportTraceServiceRequest traceRequest, @NonNull String projectName,
+            @NonNull String userName, @NonNull String workspaceName, @NonNull String workspaceId) {
+
+        var opikSpans = traceRequest.getResourceSpansList().stream()
+                .flatMap(resourceSpans -> resourceSpans.getScopeSpansList().stream())
+                .flatMap(scopeSpans -> scopeSpans.getSpansList().stream())
+                .map(OpenTelemetryMapper::toOpikSpan)
+                .map(opikSpan -> opikSpan.toBuilder()
+                        .projectName(projectName)
+                        .createdBy(userName)
```

Comment:
I just remembered that you don't need to set it; the value will be extracted from the context. This is to ensure we use the right user name